pallets/automation-time/src/tests.rs

```diff
@@ -309,6 +309,67 @@ fn schedule_xcmp_errors_bad_origin() {
 	})
 }
 
+#[test]
+fn schedule_auto_compound_delegated_stake() {
+	let alice = AccountId32::new(ALICE);
+	let bob = AccountId32::new(BOB);
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		get_funds(alice.clone());
+		assert_ok!(AutomationTime::schedule_auto_compound_delegated_stake_task(
+			Origin::signed(alice.clone()),
+			SCHEDULED_TIME,
+			3_600,
+			bob.clone(),
+			1_000_000_000,
+		));
+		let task_id = AutomationTime::get_scheduled_tasks(SCHEDULED_TIME)
+			.expect("Task should be scheduled")[0];
+		assert_eq!(
+			AutomationTime::get_task(task_id),
+			Some(Task::<Test>::create_auto_compound_delegated_stake_task(
+				alice.clone(),
+				AutomationTime::generate_auto_compound_delegated_stake_provided_id(&alice, &bob),
+				SCHEDULED_TIME,
+				3_600,
+				bob,
+				1_000_000_000,
+			))
+		);
+	})
+}
+
+#[test]
+fn schedule_auto_compound_with_bad_frequency() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		assert_noop!(
+			AutomationTime::schedule_auto_compound_delegated_stake_task(
+				Origin::signed(AccountId32::new(ALICE)),
+				SCHEDULED_TIME,
+				4_000,
+				AccountId32::new(BOB),
+				100_000,
+			),
+			Error::<Test>::InvalidTime,
+		);
+	})
+}
+
+#[test]
+fn schedule_auto_compound_with_high_frequency() {
```

Comment:
might not need, since we test for this in the is_valid_time, which should already have been tested earlier